### PR TITLE
clean CMake for mobile

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

1) corrected potential confusing spelling error in build.sh 
2) Added CMakeList.txt files to root directories that did not have one.

This PR makes the use of CMakeLists.txt consistent (each root directory have its own
CMakeLists.txt) and at the same time cleans the central CMakeLists.txt

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # in subsurface/build)
 #
 # by default it puts the build folders in
-# ./subsurface/llibdivecomputer/build (libdivecomputer build)
+# ./subsurface/libdivecomputer/build (libdivecomputer build)
```
Comment (dirkhh) [inline scripts/build.sh:15]: not sure this qualifies as "disturbing" ("causing anxiety; worrying"), but yeah, typo.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # cmake based build of Subsurface
 
+
```
Comment (dirkhh) [inline CMakeLists.txt]: why are we adding another new line here? That seems like leftover from some other changes maybe?

File: map-widget/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# map widget to show locations
+
+# library used by mobile build
+set(SUBSURFACE_MAPWIDGET_SRCS
+	qmlmapwidgethelper.cpp
+)
```
Comment (dirkhh) [inline map-widget/CMakeLists.txt:6]: This is truly just a question... but... is it worth adding this for one file? In other words, besides logical consistency and what not, what do we gain by doing this?

File: desktop-widgets/CMakeLists.txt [hunk 4]
```diff
@@ -55,7 +55,6 @@ set (SUBSURFACE_UI
 
 # the interface, in C++
 set(SUBSURFACE_INTERFACE
-	../map-widget/qmlmapwidgethelper.cpp
```
Comment (dirkhh) [inline desktop-widgets/CMakeLists.txt:58]: shouldn't this be part of the last commit? Aren't we getting a compile error otherwise (haven't tried)

**Review by dirkhh — APPROVED** [review]
I understand that this is cleaner. I'm not sure that's where I would have focused my time.
It doesn't seem to break anything, so "approved"...

File: scripts/build.sh (hunk 1 above)
Comment (janiversen) [reply scripts/build.sh:15]: updated text in PR (it is not present in the commit text).

File: CMakeLists.txt (hunk 2 above)
Comment (janiversen) [reply CMakeLists.txt]: Corrected.

File: map-widget/CMakeLists.txt (hunk 3 above)
Comment (janiversen) [reply map-widget/CMakeLists.txt:6]: That argument is valid for all CMakeList.txt files in subdirectories. Why not just make one big central file (e.g. mobile-widgets, contained a lot of entries in the central file).

We do not gain performance or anything, but consistency means easier to understand and thereby easier to maintain (at least for me). The other gain is that each root directory is a library.

And I "believe" that in a short future, there will be a PR with a dynamic map that allows the user to change locations (at least the current experiments looks good).

File: desktop-widgets/CMakeLists.txt (hunk 4 above)
Comment (janiversen) [reply desktop-widgets/CMakeLists.txt:58]: I build and test every single commit, so there are no compile errors and it works.

I have no idea how CMake deals with the conflict, but please remember the old central CMake reference qmlmapwidgethelper.cpp, so this problem have existed for a long time.

**Review by dirkhh — APPROVED** [review]

**Pull request merged**